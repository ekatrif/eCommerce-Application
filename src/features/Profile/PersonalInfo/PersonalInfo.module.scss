.profile__personal {
  display: grid;
  grid-template-columns: repeat(auto-fill, max(180px, 15vw));
  gap: 10px;
}

.profile__item {
  padding: 1rem;
  border-radius: 10px;
  background-color: #e9e7e7;

  &__title {
    font-size: 1.3rem;
  }

  &__value {
    margin-top: 10px;
  }
}

@media (max-width: 900px) {
  .profile__personal {
    grid-template-columns: 1fr 1fr;
    row-gap: 10px;
  }
}

@media (max-width: 420px) {
  .profile__personal {
    grid-template-columns: 1fr;
  }
}