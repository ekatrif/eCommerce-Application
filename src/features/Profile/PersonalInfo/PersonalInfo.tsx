import classes from './PersonalInfo.module.scss';

interface PersonalInfoProps {
  firstName: string;
  lastName: string;
  birthDate: string;
  email: string;
}

const PersonalInfo: React.FC<PersonalInfoProps> = ({ firstName, lastName, birthDate, email }) => {
  return (
    <div className={classes.profile__personal}>
      <div className={classes.profile__item}>
        <p className={classes.profile__item__title}>FIRST NAME</p>
        <p className={classes.profile__item__value}>{firstName}</p>
      </div>
      <div className={classes.profile__item}>
        <p className={classes.profile__item__title}>LAST NAME</p>
        <p className={classes.profile__item__value}>{lastName}</p>
      </div>
      <div className={classes.profile__item}>
        <p className={classes.profile__item__title}>BIRTH DATE</p>
        <p className={classes.profile__item__value}>{birthDate}</p>
      </div>
      <div className={classes.profile__item}>
        <p className={classes.profile__item__title}>EMAIL</p>
        <p className={classes.profile__item__value}>{email}</p>
      </div>
    </div>
  );
};

export default PersonalInfo;
