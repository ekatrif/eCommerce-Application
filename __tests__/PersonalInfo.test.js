import React from 'react';
import renderer from 'react-test-renderer';
import PersonalInfo from '../src/features/Profile/PersonalInfo/PersonalInfo';

test('PersonalInfo renders correctly', () => {
  const tree = renderer
    .create(<PersonalInfo />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
