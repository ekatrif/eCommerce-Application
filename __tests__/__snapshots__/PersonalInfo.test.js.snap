// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalInfo renders correctly 1`] = `
<div
  className="profile__personal"
>
  <div
    className="profile__item"
  >
    <p
      className="profile__item__title"
    >
      FIRST NAME
    </p>
    <p
      className="profile__item__value"
    />
  </div>
  <div
    className="profile__item"
  >
    <p
      className="profile__item__title"
    >
      LAST NAME
    </p>
    <p
      className="profile__item__value"
    />
  </div>
  <div
    className="profile__item"
  >
    <p
      className="profile__item__title"
    >
      BIRTH DATE
    </p>
    <p
      className="profile__item__value"
    />
  </div>
  <div
    className="profile__item"
  >
    <p
      className="profile__item__title"
    >
      EMAIL
    </p>
    <p
      className="profile__item__value"
    />
  </div>
</div>
`;
